src/Shared/ServerInfrastructure/StringUtilities.cs

```diff
@@ -387,28 +402,54 @@ public static string ConcatAsHexSuffix(string str, char separator, uint number)
                 length += str.Length;
             }
 
-            return string.Create(length, (str, separator, number), (buffer, tuple) =>
+            return string.Create(length, (str, separator, number), s_populateSpanWithHexSuffix);
+        }
+
+        private static void PopulateSpanWithHexSuffix(Span<char> buffer, (string str, char separator, uint number) tuple)
```

Comment:
I'll start a new comment to keep it clean.

----

I have a variant with SWAR

<details>
  <summary>SWAR code</summary>

```c#
[MethodImpl(MethodImplOptions.AggressiveInlining)]
private static void Number2HexSwar(int number, ref char dest)
{
    Debug.Assert(Bmi2.X64.IsSupported);

    ulong input = (ulong)(uint)number;      // cast to uint avoids movsxd and uses mov
    ulong isolatedNibbles = Bmi2.X64.ParallelBitDeposit(input, 0x0F0F0F0F_0F0F0F0Ful);

    Debug.Assert((isolatedNibbles & 0xF0F0F0F0_F0F0F0F0ul) == 0);

    // JIT will treat these as constants
    ulong packedAscii0 = Packed('0');
    ulong packedCorrection = Packed(0x80 - ('9' - '0' + 1));
    ulong packed0x80 = Packed(0x80);
    ulong packedHexShiftToAF = Packed('A' - '0' - 10);

    ulong asciiHexNumbers = isolatedNibbles + packedAscii0;
    ulong asciiHexChars = isolatedNibbles + packedCorrection;
    ulong flag = asciiHexChars & packed0x80;
    ulong hexAFMask = flag - (flag >> 7);

    ulong hex = asciiHexNumbers + (hexAFMask & packedHexShiftToAF);
    hex = BinaryPrimitives.ReverseEndianness(hex);

    ulong hex0 = hex;
    ulong hex1 = hex >> 32;

    Unsafe.Add(ref Unsafe.As<char, ulong>(ref dest), 0) = Bmi2.X64.ParallelBitDeposit(hex0, 0x00FF00FF_00FF00FFul);
    Unsafe.Add(ref Unsafe.As<char, ulong>(ref dest), 1) = Bmi2.X64.ParallelBitDeposit(hex1, 0x00FF00FF_00FF00FFul);

    static ulong Packed(ulong a) => a * 0x01010101_01010101ul;
}
```
</details>
In the microbenchmarks it is ~20...30% slower than the SIMD-variant. But it has no memory access, so maybe it is a good choice for real-world scenarios.
@benaadams can you please run your benchmarks with this? I don't have a setup for this here.